### PR TITLE
Issue59

### LINKED ISSUE #59: merTools is unaware of user specified formulas
Thanks to @cchoirat for reporting this:

``` r
set.seed(271828)
data(sleepstudy)
```

``` r
fm1 <- lmer(Reaction ~ Days + (Days|Subject), data=sleepstudy)
draw(fm1, type = 'average')
```

```
Number of observations < 20, random effect quantiles may not be well-defined.
  Reaction Days Subject
1 298.5079  4.5     308
```

``` r
fm2 <- lmer(Reaction ~ Days + I(Days^2) + (Days|Subject), data=sleepstudy)
draw(fm2, type = 'average')
```

```
Number of observations < 20, random effect quantiles may not be well-defined.
  Reaction Days I(Days^2) Subject
1 298.5079  4.5      28.5     308
```

I(Days^2) is 28.5.  It should be 20.25.

Right now `draw` is unaware of any formulas and is independently computing the mean of the `Days` and the `I(Days^2)` vectors, which is causing this issue.

### PR DESCRIPTION
**Pull request by carlbfrederick**
This should fix #59 so that draw() will attempt to calculate average obs values correctly if possible or issue a warning that returned averages may not be appropriate and can be fixed by supplying untransformed data.

Still need to write some tests (I can adapt the toy models I used to work on this function) and decide if we want to export `averageObs()`, `randomObs()`, and `findFormFuns()` ... and perhaps include an example with origData specified.

### CODE REVIEW

**Comment by carlbfrederick** [comment]
I am planning to get to them this week, so if you are ok to wait till the weekend we can push to CRAN then.

**Comment by jknowles** [comment]
Yes perfect. That should stop the recurring e-mails from Prof. Ripley :+1:

**Comment by carlbfrederick** [comment]
Ok, wrote some unit tests for findFormFuns() and it's parent functions to ensure that warnings are passed appropriately, all passed with SHA: 54441cd6

@jknowles this should boost our coverage, so feel free to review and merge when you have a chance. This will close #59

File: tests/testthat/test_merData.R [hunk 1]
```diff
@@ -384,3 +384,79 @@ test_that("Nested specifications work", {
...
+  expect_length(findFormFuns(s9), 7L)
+  ##. Scenario 10: two columns in I with no main effects
+  s10 <- lmer(y ~ I(log(a) + b^3) + c + d + (1|grp), data=play)
+  expect_warning(findFormFuns(s10))
+  expect_false(suppressWarnings(findFormFuns(s10)$`I(log(a) + b^3)`) == log(trueMeans$a) + trueMeans$b^3)
+  expect_silent(findFormFuns(s10, origData = play))
+  expect_equal(findFormFuns(s10, origData = play)$`I(log(a) + b^3)`, log(trueMeans$a) + trueMeans$b^3)
+  ##. Test that draw, draw.merMod and averageObs accept origData and issue warning if appropriate
+  expect_warning(averageObs(s10))
+  expect_silent(averageObs(s10, origData = play))
```
Comment (jknowles) [inline tests/testthat/test_merData.R]: Great test suite, this looks perfect.

**Comment by jknowles** [comment]
Great work @carlbfrederick -- this is beyond what I expected we'd be able to do. Very pleased.

**Pull request merged**